stratum/hal/lib/common/hal.cc

```diff
@@ -329,6 +337,15 @@ ::util::Status Hal::RegisterSignalHandlers() {
     }
     old_signal_handlers_[s] = h;
   }
+  // Create the pipe to transfer signals.
+  int pipe_fds[2];
+  CHECK_RETURN_IF_FALSE(pipe(pipe_fds) == 0) << "Could not create pipe.";
+  pipe_read_fd_ = pipe_fds[0];
+  pipe_write_fd_ = pipe_fds[1];
+  // Start the signal waiter thread that initiates shutdown.
+  CHECK_RETURN_IF_FALSE(pthread_create(&signal_waiter_tid_, nullptr,
+                                       SignalWaiterThreadFunc, nullptr) == 0)
+      << "Could not start the signal waiter thread.";
```

Comment:
```suggestion
      << "Could not start the signal handling thread.";
```
Maybe?

Reply:
Then we should also rename `signal_waiter_tid_` and `SignalWaiterThreadFunc` to keep it consistent.

Reply:
On second thought, I think signal waiter makes more sense. Let's leave this as is.